### ISSUE TITLE
Spec: KubeVirt Integration

### ISSUE THREAD

**Issue by nerdengineering** [networkservicemesh/networkservicemesh#815]
## Overview
I would like to explore the integration of NSM with KubeVirt. In addition to VIM integration, being able to take advantage of KubeVirt's ability to have K8s schedule VMs while NSM provides a robust L2/L3 networking model leads to strong long term potential. 

NSM + KubeVirt would allow for the use of the VM resource type, enabling declaratively scheduled VMs in K8s, as well as mitigate some of the restrictions the VM currently inherits through the K8s networking model. 

The long term implication being that K8s not only handles containers for end-users but also VMs, potentially replacing the need for additional VIMs and their 3rd party networking intermediaries such as NSX/Neutron/etc.

## Link to the Google Docs document
Place holder: https://docs.google.com/document/d/1V2IF7FWVT2BitWG5e9PWVaj1pUCwsn3XHYONoKxbXFk/edit

## References
https://github.com/kubevirt/kubevirt

https://github.com/kubevirt/ovs-cni

**Comment by nickolaev** [comment]
I can see [vhost-user](http://docs.openvswitch.org/en/latest/topics/dpdk/vhost-user/) sitting behind the corner here :)

**Comment by nerdengineering** [comment]
I shouldn't have called out KubeVirt specifically to be honest. The spec is
more specifically looking to explore K8s scheduled VMs with NSM plumbed
networking. KubeVirt was just the plugin that first jumped into my mind.

**Comment by edwarnicke** [comment]
@nerdengineering Do you want to preserve this against the merciless onslaught of stale-bot? :)

**Comment by nerdengineering** [comment]
Yes, I'm sorry for the delay on this, just a lot of other stuff going on. I'll begin work on this once I tie up a few other things with Frederick.

**Comment by edwarnicke** [comment]
@nerdengineering All good, just didn't want stale-bot reaping it :)
